### PR TITLE
feat: Create project python API

### PR DESCRIPTION
**Pull request by rouk1**
Skore project can now be created with a public python API.

Usage: 

```python
import skore

p = skore.create("my_project.skore")
```

See docstring for advanced usage.

### CODE REVIEW

File: skore/src/skore/project.py
Comment (thomass-dev) [inline skore/src/skore/project.py:1]: Please can you:
- split this file in three part: `project/project.py`, `project/load.py` and `project/create.py`,
- move the exception to the root `exception.py`.

I think it will add readability.

**Review by thomass-dev — CHANGES_REQUESTED** [review]
What do you think about:
- splitting the project file in three parts: `project/project.py`, `project/load.py` and `project/create.py`,
- moving the exceptions to the root exception.py.

I think it will add readability.

File: examples/plot_02_overview_skore_ui.py [hunk 1]
```diff
@@ -19,21 +21,28 @@
...
+import altair as alt
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import PIL
+import plotly.express as px
+from sklearn.datasets import load_diabetes
+from sklearn.linear_model import Lasso
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
```
Comment (thomass-dev) [inline examples/plot_02_overview_skore_ui.py:37]: Please revert these changes.

Comment (rouk1) [reply examples/plot_02_overview_skore_ui.py]: Done.

Comment (thomass-dev) [reply examples/plot_02_overview_skore_ui.py]: No its not, sorry. Check the diff.

**Comment by rouk1** [comment]
> What do you think about:
> 
>     * splitting the project file in three parts: `project/project.py`, `project/load.py` and `project/create.py`,
> 
>     * moving the exceptions to the root exception.py.
> 
> 
> I think it will add readability.

Done

File: examples/plot_02_overview_skore_ui.py (hunk 1 above)
Comment (rouk1) [reply examples/plot_02_overview_skore_ui.py]: My bad I did not understood correctly. It should be done now.

File: examples/plot_01_getting_started.py [hunk 2]
```diff
@@ -53,9 +50,6 @@
 # Let us load the project and add an integer to it for example:
```
Comment (thomass-dev) [inline examples/plot_01_getting_started.py:50]:
```suggestion
```

File: examples/plot_02_overview_skore_ui.py [hunk 3]
```diff
@@ -11,29 +11,25 @@
...
+"""
+.. _example_overview_skore_ui:
+
+========================
+Overview of the skore UI
+========================
+
+This example provides an overview of the functionalities and the different types
+of items that you can store in a skore :class:`~skore.Project`.
+"""
```
Comment (thomass-dev) [inline examples/plot_02_overview_skore_ui.py:23]: Bad merge-conflict resolution, it's a duplicate of the file-header.

Comment (thomass-dev) [inline examples/plot_02_overview_skore_ui.py:23]: Bad merge-conflict resolution, to revert.

File: examples/plot_03_cross_validate.py [hunk 4]
```diff
@@ -11,31 +11,19 @@
 """
 
 # %%
-# Creating and loading the skore project
-# ======================================
-#
-# We start by creating a temporary directory to store our project such that we can
-# easily clean it after executing this example. If you want to keep the project,
-# you have to skip this section.
```
Comment (thomass-dev) [inline examples/plot_03_cross_validate.py:19]: Bad merge-conflict resolution, to revert.

File: examples/plot_03_cross_validate.py [hunk 5]
```diff
@@ -11,31 +11,19 @@
...
-import subprocess
-
-# create the skore project
-subprocess.run(
-    f"python3 -m skore create my_project_cv --working-dir {temp_dir.name}".split()
-)
-
+import matplotlib.image as mpimg
+import matplotlib.pyplot as plt
+from sklearn.model_selection import cross_validate as sklearn_cross_validate
```
Comment (thomass-dev) [inline examples/plot_03_cross_validate.py:19]: Bad merge-conflict resolution, to revert.

File: examples/plot_02_overview_skore_ui.py [hunk 6]
```diff
@@ -9,31 +9,15 @@
...
 """
 
-# %%
-# Creating and loading a skore project
-# ====================================
-#
-# We start by creating a temporary directory to store our project such that we can
-# easily clean it after executing this example. If you want to keep the project,
-# you have to skip this section.
 import tempfile
```
Comment (thomass-dev) [inline examples/plot_02_overview_skore_ui.py:12]:
```suggestion
# %%
# Creating and loading a skore project
# ====================================
#
# We start by creating a temporary directory to store our project such that we can
# easily clean it after executing this example. If you want to keep the project,
# you have to skip this section.
import tempfile
```

**Review by thomass-dev — APPROVED** [review]
